src/reducers/AddSiteReducer.js

```diff
@@ -148,8 +149,11 @@ export default (state = INITIAL_STATE, action) => {
         case ADD_SITE_FAILURE:
             return {...state};
 
-        case INITIALIZE_MAP:
-            return {...state};
+        case MAP_IS_INITIALIZING:
+            return {...state, sitesShouldUpdate: false};
+
+        case FACEBOOK_LOGOUT_COMPLETE:
+            return INITIAL_STATE;
```

Comment:
When we upload a new site, we fire off a `sitesShouldUpdate: true` to tell the `SearchScreen` to kick off a new search and get the latest results. Once that fetch is initialized, we now fire off the `MAP_IS_INITIALIZING` type, which is captured here, and we turn the `sitesShouldUpdate` flag back to `false`.